addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1814,14 +1860,25 @@ var SnippetsMenu = Widget.extend({
             editor.removeSnippet();
         });
     },
+    /**
+     * @private
+     */
+    _onSnippetCloned: function (ev) {
+        if (ev.data.$target.hasClass('o_snippet_invisible')) {
+            this._addInvisibleElementsToLeftPanel(ev.data.$target);
+        }
+    },
     /**
      * Called when a snippet is removed -> checks if there is draggable snippets
      * to enable/disable as the DOM changed.
      *
      * @private
      */
-    _onSnippetRemoved: function () {
+    _onSnippetRemoved: function (ev) {
         this._disableUndroppableSnippets();
+        if (ev.target.$target.hasClass('o_snippet_invisible')) {
+            $('#' + ev.target.$target.attr('invisiblePanelId')).remove();
```

Comment:
You probably should not have to mark the DOM and I am sure you should not have to search the DOM for the UI element you created. You should have a map somewhere mapping the DOM elements to your UI elements